File: Data Structures & Algorithms/trapping-rain-water/submission-2.cpp
class Solution {
public:
    int trap(vector<int>& height) {
        int n = height.size();
        vector<int> left(n,0);
        vector<int> right(n,0);
        int res = 0;
        int maxl = height[0];
        for(int i=1;i<n;i++){
            left[i] = maxl;
            maxl = max(maxl,height[i]);
        }
        int maxr = height[n-1];
        for(int i=n-2;i>=0;i--){
            right[i] = maxr;
            maxr = max(maxr,height[i]);
        }
        for(int i=1;i<n-1;i++){
            int m = min(left[i],right[i]) - height[i];
            if(m>0) res+=m;
        }
        return res;
    }
};

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-2.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        l = 0
        r = 0
        n = len(nums)
        sumv = 0
        res = n+1
        while r<n:
            sumv += nums[r]
            while l<=r and sumv>=target:
                res = min(res, r-l+1)
                sumv -= nums[l]
                l+=1
            r+=1
        if res == n+1: return 0
        return res